src/services/UserService.js

```diff
@@ -0,0 +1,90 @@
+import { $db } from '@/services/firebase'
+import { configService } from '@/services/Config.service.js'
+
+const COLLECTION_NAME = 'user'
+
+class UserService {
+  constructor () {
+    this.user = null
+  }
+
+  async signIn ({ firstName, lastName, initials, masterPassword }) {
+    const loadedPassword = await configService
+      .fetchMasterPassword()
+    if (loadedPassword !== masterPassword) {
+      return true
+    } else {
+      await this.fetchUserByInitials(initials)
+        .then(user => {
+          if (user) {
+            return user
+          } else {
+            const user1 = {
+              firstName,
+              lastName,
+              initials
+            }
+            return $db().collection(COLLECTION_NAME).add(user1).then(response => {
+              user1.id = response.id
+              return user1
+            })
+          }
+        })
```

Comment:
Könnte man das nicht auf async await umschreiben?
```suggestion
      const user = await this.fetchUserByInitials(initials)
          if (user) {
            return user
          } else {
            const newUser = {
              firstName,
              lastName,
              initials
            }
            const newUserFromDb = $db().collection(COLLECTION_NAME).add(newUser)
              newUser.id = newUserFromDb.id
              return newUser
          }
```